### DISCUSSION TITLE
organizzazione fisica - 12

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#91] [Esercizi]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/ff107114-4c9f-4cca-a6a4-970a1ce824c4)

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/c10ca133-4bee-4e70-b71e-40f130f966e9)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/38bf3e34-9e60-4722-9933-5f260c8d1ed4)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/d15839ba-7d48-477e-a2e0-2e15d3aaea17)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/5edf97bf-3688-474d-9ec3-3bbae578a8e7)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/f1ad6718-1b17-42f1-b871-dfec8c714f89)

**Reply by IonutCicio** [reply]
Al punto b), gli accessi sono 4 se il livello 4 (LVL 4) è già in memoria. Noi teniamo solo la radice in memoria al massimo, non il file principale con tutti i record _(non basterebbe la RAM)_. Il calcolo al punto c) pare un po' strano...

**Reply by Warcophyr** [reply]
correzione ultimo pezzo del punto b) e correzione punti c) 
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/2db2fbea-3547-4306-b535-3ea6b6f8f400)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/537bff4e-0a5e-4c5b-8267-19d684ccbf3c)

**Comment by IonutCicio** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/18371125-e713-4992-bbbd-7facc4fc6330)

**Comment by Hiperversum** [comment]
Ragazzi, l'esercizio chiede al punto a) di considerare il minimo numero di blocchi nel file indice. Dunque non dovete considerare i blocchi pieni a metà. Per avere il minimo numero di blocchi allora devono esserci il massimo numero di record per blocco. Quindi dovete considerare il blocco per intero

**Reply by IonutCicio** [reply]
Dovrei ricontrollare un attimo, può allegare uno screenshot delle slide che riguardano questo tipo di esercizi o un link ad un esercizio simile (con il minimo numero di blocchi nell'indice) ma fatto bene? Effettivamente quei 27 puntatori per blocco (nell'indice) sono un po' pochi rispetto a quanti se ne potrebbero mettere, ma ricordo anche che il B-tree ha delle regole un po' particolari per questo tipo di calcoli. La sua osservazione è ragionevole, ma qualche spiegazione in più sarebbe gradita, sarebbe ottimo se allegasse la sua soluzione.

_(le ammollo il lavoro di verifica perché, nonostante ci tenga alla correzione, in questo periodo questo tipo di attività non è fra le mie priorità)_

**Comment by alemilos** [comment]
<img width="554" height="855" alt="Screenshot 2026-04-11 at 10 21 09" src="https://github.com/user-attachments/assets/7c50ad07-915b-4f07-a07d-12876ba2cf6e" />
